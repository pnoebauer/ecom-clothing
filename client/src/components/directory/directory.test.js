import React from 'react';
import {shallow} from 'enzyme';

import {Directory} from './Directory.component.jsx';
import MenuItem from '../menu-item/menu-item.component';

const mockSections = [
	{
		title: 'hats',
		imageUrl: 'https://i.ibb.co/cvpntL1/hats.png',
		id: 1,
		linkUrl: 'shop/hats',
	},
	{
		title: 'jackets',
		imageUrl: 'https://i.ibb.co/px2tCc3/jackets.png',
		id: 2,
		linkUrl: 'shop/jackets',
	},
];

describe('<Directory /> testing', () => {
	test('render Directory component', () => {
		expect(shallow(<Directory sections={mockSections} />)).toMatchSnapshot();
	});

	test('renders a MenuItem component for each element in the provided sections array ', () => {
		const expectedNumber = mockSections.length;
		const wrapper = shallow(<Directory sections={mockSections} />);
		// const result = wrapper.find(MenuItem);
		// console.log(wrapper.debug());
		// console.log(result.debug());

		expect(wrapper.find(MenuItem)).toHaveLength(expectedNumber);
	});
});
