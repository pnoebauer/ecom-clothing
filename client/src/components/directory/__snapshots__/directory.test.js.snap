// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Directory /> testing render Directory component 1`] = `
<directorystyles__DirectoryMenuContainer>
  <withRouter(MenuItem)
    imageUrl="https://i.ibb.co/cvpntL1/hats.png"
    key="1"
    linkUrl="shop/hats"
    title="hats"
  />
  <withRouter(MenuItem)
    imageUrl="https://i.ibb.co/px2tCc3/jackets.png"
    key="2"
    linkUrl="shop/jackets"
    title="jackets"
  />
</directorystyles__DirectoryMenuContainer>
`;
